example/declarative.dart

```diff
@@ -1,29 +1,31 @@
 // Copyright 2013 Google. All rights reserved. Use of this source code is
 // governed by a BSD-style license that can be found in the LICENSE file.
 
-library grinder_example_1;
+library grinder.example.declarative;
 
 import 'package:grinder/grinder.dart';
 
-void main([List<String> args]) {
-  task('init', init);
-  task('compile', compile, ['init']);
-  task('deploy', deploy, ['compile']);
-  task('default', null, ['deploy']);
-
+void main(List<String> args) {
+  addAnnotatedTasks();
   startGrinder(args);
 }
 
-void init(GrinderContext context) {
+@Task()
+init(GrinderContext context) {
   context.log("I set things up");
 }
 
-void compile(GrinderContext context) {
+@Task(depends: const ['init'])
+compile(GrinderContext context) {
   context.log("I'm compiling now...");
   //context.fail('woot');
 }
 
-void deploy(GrinderContext context) {
+@Task(depends: const ['compile'])
```

Comment:
Another annotation scheme:
```
@task
@depends('foo', 'bar');
@description('My nice task description.');
deploy(GrinderContext context) {
...
}
```

It doesn't have the benefit of being all wrapped up in one annotation, but is potentially nicer to read. I don't have a preference but want to bring up other formats.

The new `rpc` package (https://github.com/dart-lang/rpc) uses a single annotation. When writing code that uses it I don't think the `description` is as nice to read. I'd love to be able to just use the dartdoc for the method but that's not available at runtime -